src/test/java/com/google/sps/servlets/UserDataServletTest.java

```diff
@@ -195,22 +192,22 @@ public void testPostCreateUserInfo() throws Exception {
     servletUnderTest.doPostWrapper(datastore, mockRequest, mockResponse);
 
     // Query the local datastore for the newly created user entity
-    PreparedQuery results = datastore.prepare(new Query(TEST_USER_ID));
-    Assert.assertEquals(1, results.countEntities());
-    Entity userEntity = results.asSingleEntity();
+    Entity userEntity = datastore.prepare(new Query(TEST_USER_ID)).asSingleEntity();
+    assertThat(userEntity).isNotNull();
 
     // Verify that the user entity has the correct properties
-    Assert.assertEquals((String) userEntity.getProperty(USER_ID_PROPERTY), TEST_USER_ID);
-    Assert.assertEquals((String) userEntity.getProperty(USER_NAME_PROPERTY), TEST_USER_NAME);
-    Assert.assertEquals((String) userEntity.getProperty(USER_EMAIL_PROPERTY), TEST_USER_EMAIL);
-    Assert.assertEquals((String) userEntity.getProperty(USER_BIO_PROPERTY), TEST_USER_BIO);
-    Assert.assertEquals(((ArrayList<String>) userEntity.getProperty(USER_FRIENDS_LIST_PROPERTY)).toArray(),
-        TEST_USER_FRIENDS_LIST);
+    assertThat((String) userEntity.getProperty(USER_ID_PROPERTY)).contains(TEST_USER_ID);
+    assertThat((String) userEntity.getProperty(USER_NAME_PROPERTY)).contains(TEST_USER_NAME);
+    assertThat((String) userEntity.getProperty(USER_EMAIL_PROPERTY)).contains(TEST_USER_EMAIL);
+    assertThat((String) userEntity.getProperty(USER_BIO_PROPERTY)).contains(TEST_USER_BIO);
+    assertThat(Arrays.toString(((ArrayList<String>) userEntity.getProperty(USER_FRIENDS_LIST_PROPERTY)).toArray()))
```

Comment:
Why is this going to Strings (I think there's a second usage of this pattern lower)? It seems to me something like this would work:

```
assertThat((ArrayList<String>) userEntity.getProperty(USER_FRIENDS_LIST_PROPERTY).containsExactly(Arrays.toList(TEST_USER_FRIENDS_LIST);
```